src/Hamiltonians/HubbardRealSpace.jl

```diff
@@ -113,64 +113,109 @@ It is implemented recursively to ensure type stability.
     return self + row + _interactions(as, rest)
 end
 
+"""
+    trap_potential(add::SingleComponentFockAddress, pot)
+
+Calculate trap potential energy of a single-component address `add`. The (precomputed) 
+potential energy per particle at each point in the lattice is given by the vector `pot`.
```

Comment:
```suggestion
    trap_potential(add::AbstractFockAddress, pot)

Calculate the value of a diagonal single particle operator (e.g. a trap potential) at 
the address `add`. The (precomputed) potential energy per particle at each mode
passed as `pot` should be a vector for a [`SingleComponentFockAddress`](@ref), 
or a `M×C` matrix, where `M` is the number of modes and `C` the number of 
components for a [`CompositeFS `](@ref).
```math
\\sum_{iσ} V_{iσ} n_{iσ}
```
```